builder/swig/python.bzl

```diff
@@ -0,0 +1,172 @@
+        args = ["-I" + h.dirname] + args
+
+    ctx.actions.run(
+        inputs = depset([ctx.file.interface] + ctx.files.includes, transitive = [
+                    ctx.attr.lib[CcInfo].compilation_context.headers,
+                    ctx.attr._swig.data_runfiles.files,
+                ]),
+        outputs = [wrap_src, wrap_py] + swig_headers,
+        executable = ctx.file._swig,
+        arguments = args,
+    )
+
+    lib_compilation_context = ctx.attr.lib[CcInfo].compilation_context
+    compilation_context = cc_common.create_compilation_context(
+        headers = depset(ctx.attr._python_header[CcInfo].compilation_context.headers.to_list() + swig_headers, transitive = [lib_compilation_context.headers]),
+        defines = lib_compilation_context.defines,
+        framework_includes = lib_compilation_context.framework_includes,
+        includes = lib_compilation_context.includes,
+        local_defines = lib_compilation_context.local_defines,
+        quote_includes = lib_compilation_context.quote_includes,
+        system_includes = depset(
+            [file.dirname for file in ctx.attr._python_header[CcInfo].compilation_context.headers.to_list()],
+            transitive = [lib_compilation_context.system_includes],
+        )
+    )
+
+    return [
+        DefaultInfo(files = depset([wrap_src, wrap_py])),
+        CcInfo(
+            compilation_context = compilation_context,
+            linking_context = ctx.attr.lib[CcInfo].linking_context,
+        ),
+    ]
+
+
+_swig_python_wrapper = rule(
+    implementation = _swig_python_wrapper_impl,
+    attrs = {
+        "lib": attr.label(
+            doc = "The C library for which to generate the wrapper",
+            providers = [CcInfo],
+            mandatory = True,
+        ),
+        "class_name": attr.string(
+            doc = "Optional override for the python class name (default: same as target name)",
+        ),
+        "shared_lib_name": attr.string(
+            doc = "Optional override for the dynamic library name (default: '_' + target name)",
+        ),
+        "interface": attr.label(
+            doc = "Optional SWIG interface (.i) file",
+            allow_single_file = True,
+        ),
+        "includes": attr.label_list(
+            doc = "Additional SWIG files required for wrapper generation",
+            allow_files = True,
+        ),
+        "enable_cxx": attr.bool(
+            doc = "Enable SWIG C++ processing (default: False)",
+            default = True,
+        ),
+        "extra_args": attr.string_list(
+            doc = "Extra arguments to be passed to SWIG",
+        ),
+        "_python_header": attr.label(
+            default = Label("@python39//:python_headers"),
+        ),
+        "_swig": attr.label(
+            default = Label("@swig//:swig"),
+            allow_single_file = True,
+            executable = True,
+            cfg = "host",
+        ),
+    },
+)
+
+
+def swig_python_wrapper(**kwargs):
```

Comment:
Where are we passing/using a `select` in `_swig_python_wrapper`?

Reply:
Yes, we don't use `select` here, it is an obsolete comment from java.